pyproject.toml

```
@@ -2,45 +2,224 @@
 # Distributed under the terms of the Modified BSD License.
 
 [build-system]
-requires=["jupyter_packaging>=0.9,<2"]
-build-backend = "jupyter_packaging.build_api"
+requires = ["hatchling>=1.2.0"]
+build-backend = "hatchling.build"
 
-[license]
-file="LICENSE"
+[project]
+name = "jupyterlab"
+description = "JupyterLab computational environment"
+readme = "README.md"
+license = ""
+requires-python = ">=3.7"
+authors = [
+    { name = "Jupyter Development Team", email = "jupyter@googlegroups.com" },
+]
+keywords = [
+    "ipython",
+    "jupyter",
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Framework :: Jupyter",
+    "Framework :: Jupyter :: JupyterLab",
+    "Framework :: Jupyter :: JupyterLab :: 4",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Intended Audience :: System Administrators",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+]
+dependencies = [
+    "importlib-metadata>=3.6;python_version<\"3.10\"",
+    "importlib-resources>=1.4;python_version<\"3.9\"",
+    "ipykernel",
+    "jinja2>=3.0.3",
+    "jupyter_core",
+    "jupyter_server>=1.16.0,<2",
+    "jupyter_server_ydoc>=0.1.9,<0.2.0",
+    "jupyterlab_server>=2.11.1,<3",
+    "notebook_shim>=0.1",
+    "packaging",
+    "tornado>=6.1.0",
+    "tomli",
+    "ypy-websocket",
+]
+dynamic = [
+    "version",
+]
+
+[project.scripts]
+jlpm = "jupyterlab.jlpmapp:main"
+jupyter-lab = "jupyterlab.labapp:main"
+jupyter-labextension = "jupyterlab.labextensions:main"
+jupyter-labhub = "jupyterlab.labhubapp:main"
+
+[project.urls]
+Homepage = "https://jupyter.org"
+
```

Let's add more URL (ref https://pypi.org/project/links-demo/):

```suggestion
Changelog = "https://jupyterlab.readthedocs.io/en/stable/getting_started/changelog.html"
Documentation = "https://jupyterlab.readthedocs.io"
Source = "https://github.com/jupyterlab/jupyterlab"
Issues = "https://github.com/jupyterlab/jupyterlab/issues/new/choose"
Gitter = "https://gitter.im/jupyterlab/jupyterlab"
Pypi = "https://pypi.org/project/jupyterlab"

```